src/utilTypes.ts

```diff
@@ -0,0 +1,13 @@
+/* Possible return values of a sort's comparator function */
+export type ComparatorValue = 1 | -1 | 0
+
+/* A standard comparator function used within sort */
+export type ComparatorFunction<T> = (a: NonNullable<T>, b: NonNullable<T>) => ComparatorValue
+
+/* Set of file types supported for exporting thoughts */
+export type MimeType = 'text/plain' | 'text/html'
+
+/* Generic type to allow null */
+export type Nullable<T> = T | null;
+
+export type GenericObject<T = any> = {[key: string]: T}
```

Comment:
What's the advantage of using this over any? Since everything is a generic object in Javascript.

Reply:
That's true, everything is a generic object in JS and we can allow them to be like that. But this helps to narrow down the allowed value types for an object. 
Example - We can enforce the thoughtIndex to have only `thoughtObject` values - [`thoughtIndex: GenericObject<Thought>`](https://github.com/cybersemics/em/blob/5226dcfec237353436ae84973c23cacfa2b76764/src/util/initialState.ts#L33).